src/parser.ml

```diff
@@ -767,6 +768,33 @@ let blank s =
   if not (is_empty s) then raise Fail;
   Lempty
 
+let tag s =
+  match Sub.heads 2 s with
+  | ['{'; '!'] ->
+      let tag = Buffer.create 17 in
+      let rec loop s =
+        match Sub.head s with
+        | Some ':' ->
+            let _, a =
+              match Sub.head ~rev:() s with
+              | Some '}' ->
+                  attribute_string s
+              | _ ->
+                  s, Attributes.empty
+            in
+            let string_tag = Buffer.contents tag in
+            let string_tag = if string_tag = "" then "none" else string_tag in
```

Comment:
I don't think we should return `"none"` if the tag string is empty. In fact, we should probably disallow empty tags. What do you think?

Reply:
Yes, better idea, thank you